sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java

```diff
@@ -286,16 +299,27 @@ public long putLog(@NonNull Log log, @NonNull String group, @IntRange(from = Fla
                 targetToken = null;
             }
             long maxSize = mDatabaseManager.getMaxSize();
-            if (maxSize == -1) {
+            if (maxSize == OPERATION_FAILED_FLAG) {
                 throw new PersistenceException("Failed to store a log to the Persistence database.");
             }
             if (!isLargePayload && maxSize <= payloadSize) {
                 throw new PersistenceException("Log is too large (" + payloadSize + " bytes) to store in database. " +
                         "Current maximum database size is " + maxSize + " bytes.");
             }
-            contentValues = getContentValues(group, isLargePayload ? null : payload, targetToken, log.getType(), targetKey, Flags.getPersistenceFlag(flags, false));
-            long databaseId = mDatabaseManager.put(contentValues, COLUMN_PRIORITY);
-            if (databaseId == -1) {
+            int priority = Flags.getPersistenceFlag(flags, false);
+            contentValues = getContentValues(group, isLargePayload ? null : payload, targetToken, log.getType(), targetKey, priority);
+            Long databaseId = null;
+            while (databaseId == null) {
```

Comment:
We would also need to check if we have space to store a large payload file.
Draft solution:
```
while (isLargePayload && payloadSize + mLargePayloadsSize + mDatabaseManager.getCurrentSize() > maxSize)
{
    if (deleteTheOldestLog(COLUMN_PRIORITY, priority) == OPERATION_FAILED_FLAG) {
        throw new PersistenceException("no space for the large payload");
    }
}
```